### DISCUSSION TITLE
Suggestion -->Re-orderable list control

### DISCUSSION THREAD

**Discussion by aedhcarrick** [flet-dev/flet#824] [Ideas]
I wrote a kind of hack-y version for a project I'm doing, it works okay, but, every time I see the code, it makes me feel like I have a pimple. Is it feasible to port something like this? ---> https://pub.dev/packages/flutter_reorderable_list
I'm sure I'm not the only one who could find a use for it..

**Comment by FeodorFitsner** [comment]
I think you can easily do such a control with Flet by looking at this example: https://github.com/flet-dev/examples/blob/main/python/controls/drag-and-drop/drag-drop-ordering.py

**Reply by CoutinhoElias** [reply]
How would I do this example to load the data from my database or from json?

**Reply by CoutinhoElias** [reply]
In this project I create 4 columns dynamically based on the json contained in the requested_rd.py file

See my code:
[https://gist.github.com/CoutinhoElias/0db8828ef525a4e4b0e2ea3e6e4e7a2e](url)

I drag the first column to the place of the last. (It works)
I drag the second column to the place of the last one. (Does not work)

It presents an error before and after releasing the column.
#---------------------------------------------------------------- -------------------------------------------------- ----
# First error

Traceback (most recent call last):
   File "C:\Users\SV\AppData\Local\Programs\Python\Python39\lib\threading.py", line 950, in _bootstrap_inner
     self.run()
   File "C:\Users\SV\AppData\Local\Programs\Python\Python39\lib\threading.py", line 888, in run
     self._target(*self._args, **self._kwargs)
   File "c:\Users\SV\Documents\codes_sv\ft_teste\dd_personalizado.py", line 254, in drag_will_accept
     self.page.update()
AttributeError: 'NoneType' object has no attribute 'update'
Happens on line 206
Exception in thread Thread-52:

#---------------------------------------------------------------- -------------------------------------------------- ----
# Second error

Traceback (most recent call last):
   File "C:\Users\SV\AppData\Local\Programs\Python\Python39\lib\threading.py", line 950, in _bootstrap_inner
     self.run()
   File "C:\Users\SV\AppData\Local\Programs\Python\Python39\lib\threading.py", line 888, in run
     self._target(*self._args, **self._kwargs)
   File "C:\Users\SV\kivy_venv\lib\site-packages\flet_core\event_handler.py", line 28, in __sync_handler
     h(r)
   File "c:\Users\SV\Documents\codes_sv\ft_teste\dd_personalizado.py", line 228, in drag_accept
     src = self.page.get_control(e.src_id)
AttributeError: 'NoneType' object has no attribute 'get_control'

I would really appreciate someone's help here.

**Reply by burhansvural** [reply]
because self.page may have taken the value None
You need to check this situation with an if .... : .......;
change all self.page.update() to this => if self.page!=None: self.page.update()
change self.page.get_control(e.src_id) to => if self.page!= None : self.page.get_control(e.src_id) or changes all src = self.page.get_control(e.src_id) to => src = self.page.get_control(e.src_id) if self.page!=None else None

**Reply by CoutinhoElias** [reply]
It does not solve. It continues in the same way.

**Comment by aedhcarrick** [comment]
ty, this is helpful.
